src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -244,6 +214,100 @@ internal void InitAssemblyMethods(XmlMapping[] xmlMappings)
             return null;
         }
 
+        private static Assembly? LoadAssemblyByName(AssemblyName name)
+        {
+            Assembly? assembly = null;
+
+            try
+            {
+                assembly = Assembly.Load(name);
+            }
+            catch (Exception e)
+            {
+                if (e is ThreadAbortException || e is StackOverflowException || e is OutOfMemoryException)
```

Comment:
This entire catch is gone now since we won't need the opportunity to retry the assembly load with a partial name, so there is no need to special case these three exceptions to still throw.